### PR TITLE
simplify monitor address setting

### PR DESCRIPTION
**Pull request by guits**
this drops the following parameters:
    
- monitor_address_block
- monitor_interface
- monitor_address

The monitor address will be automatically set from `public_network` parameter.

### CODE REVIEW

File: infrastructure-playbooks/rolling_update.yml [hunk 1]
```diff
@@ -304,7 +304,7 @@
           delegate_facts: true
 
         - name: Non container | waiting for the monitor to join the quorum...
-          ansible.builtin.command: ceph --cluster "{{ cluster }}" -m "{{ hostvars[groups[mon_group_name][0]]['_current_monitor_address'] }}" quorum_status --format json
+          ansible.builtin.command: ceph --cluster "{{ cluster }}" -m "{{ _monitor_addresses[hostvars[groups['mons'][0]['inventory_hostname']]] }}" quorum_status --format json
```
Comment (clwluvw) [inline infrastructure-playbooks/rolling_update.yml:307]:
```suggestion
          ansible.builtin.command: ceph --cluster "{{ cluster }}" -m "{{ _monitor_addresses[groups['mons'][0]] }}" quorum_status --format json
```

File: infrastructure-playbooks/rolling_update.yml [hunk 2]
```diff
@@ -317,7 +317,7 @@
 
         - name: Container | waiting for the containerized monitor to join the quorum...
           ansible.builtin.command: >
-            {{ container_binary }} exec ceph-mon-{{ ansible_facts['hostname'] }} ceph --cluster "{{ cluster }}" -m "{{ hostvars[groups[mon_group_name][0]]['_current_monitor_address'] }}" quorum_status --format json
+            {{ container_binary }} exec ceph-mon-{{ ansible_facts['hostname'] }} ceph --cluster "{{ cluster }}" -m "{{ _monitor_addresses[hostvars[groups['mons'][0]['inventory_hostname']]] }}" quorum_status --format json
```
Comment (clwluvw) [inline infrastructure-playbooks/rolling_update.yml:320]:
```suggestion
            {{ container_binary }} exec ceph-mon-{{ ansible_facts['hostname'] }} ceph --cluster "{{ cluster }}" -m "{{ _monitor_addresses[groups['mons'][0]] }}" quorum_status --format json
```

File: tests/functional/all_daemons/group_vars/mons [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-create_crush_tree: True
-crush_rule_config: True
+create_crush_tree: false
+crush_rule_config: false
```
Comment (clwluvw) [inline tests/functional/all_daemons/group_vars/mons:3]: These two are making problems for the CI which doesn't have the HDD crush rule.

Comment (guits) [reply tests/functional/all_daemons/group_vars/mons:3]: they shouldn't be false

File: roles/ceph-mon/tasks/deploy_monitors.yml [hunk 4]
```diff
@@ -158,13 +158,13 @@
   ansible.builtin.command: >
     {{ ceph_monmaptool_cmd }}
     --create
-    {% for host in _monitor_addresses -%}
+    {% for name, addr in _monitor_addresses.items() -%}
     --addv
-    {{ host.name }}
+    {{ name }}
```
Comment (clwluvw) [inline roles/ceph-mon/tasks/deploy_monitors.yml:163]: This needs to be the hostname as the `inventory_hostname` might be a fqdn
```suggestion
    {{ hostvars[name]['ansible_hostname'] }}
```

**Comment by guits** [comment]
jenkins test centos-non_contanier-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-all_daemons

**Comment by guits** [comment]
jenkins test centos-container-all_daemons

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_osds

**Comment by guits** [comment]
jenkins test centos-non_container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-container-lvm_batch

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Pull request merged**